libkbfs/block_retrieval_worker_test.go

```diff
@@ -61,20 +62,15 @@ func (bg *fakeBlockGetter) getBlock(ctx context.Context, kmd KeyMetadata, blockP
 		return errors.New("Block doesn't exist in fake block map")
 	}
 	// Wait until the caller tells us to continue
-	select {
-	case <-source.ch:
-		bytes, err := bg.codec.Encode(source.block)
-		if err != nil {
-			return err
-		}
-		err = bg.codec.Decode(bytes, block)
-		if err != nil {
-			return err
+	for {
+		select {
+		case source.startCh <- struct{}{}:
```

Comment:
This way, you can wait on `startCh` as many times as you want (although just waiting once will tell you what you need to know), and then when you tell the `continueCh` to continue, the request will finish.